datasets/polsum/polsum.py

```diff
@@ -0,0 +1,192 @@
+                "authors": datasets.Value("string"),
+                "body": datasets.Value("string"),
+                "summaries": datasets.features.Sequence(
+                    {
+                        "ratio": datasets.Value("string"),
+                        "type": datasets.Value("string"),
+                        "author": datasets.Value("string"),
+                        "body": datasets.Value("string"),
+                        "spans": datasets.features.Sequence(
+                            {
+                                "start": datasets.Value("int32"),
+                                "end": datasets.Value("int32"),
+                                "span_text": datasets.Value("string"),
+                            }
+                        ),
+                    }
+                ),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepaths": glob.glob(data_dir + "/*/*/*.xml"),
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepaths):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        for i, xml_path in enumerate(filepaths):
```

Comment:
```suggestion
        for i, xml_path in enumerate(sorted(filepaths)):
```
we need to make the order deterministic here